hw/ip/gpio/data/gpio.hjson

```diff
@@ -433,5 +439,86 @@
         }
       ],
     },
+    { multireg:
+      { name: "INP_PRD_CNT_CTRL",
+        cname: "INP_PRD_CNT_CTRL",
+        desc: "Control register of one input period counter.",
+        count: "NumInpPeriodCounters"
+        swaccess: "rw",
+        hwaccess: "hrw",
+        fields: [
+          { bits: "0",
+            name: "enable",
+            resval: "0",
+            desc: '''Enable this input period counter.
+                     After enabling, this counter waits for the next sensitive edge (see `polarity` bit) of the input to start counting.
+                     After that, it counts the number of clock cycles until the next sensitive edge.
+                     Once that has happened, it stores the result in !!INP_PRD_CNT_VAL[i], which completes one measurement.
+                     Then, if the `continuous_mode` bit of this register is not set, the counter clears the `enable` bit and returns to idle (see description of the `continuous_mode` bit for what happens if that bit is set).
+                  '''
+          },
+          { bits: "1",
+            name: "continuous_mode",
+            resval: "0",
+            desc: '''Continuously count the input period.
+                     When one measurement is completed (see description of `enable` bit) and this bit is set, the internal counter is reset to zero and immediately begins counting the clock cycles until the subsequent sensitive edge of the input.
+                     The current result in !!INP_PRD_CNT_VAL[i] and the `enable` bit are not cleared.
+
+                     This bit may only be changed while the `enable` bit is `0`.
+                  '''
+          },
+          { bits: "2",
+            name: "polarity",
+            resval: "1",
+            desc: '''Polarity of this input period counter.
+                     If 0, it is sensitive to falling edges of the input.
+                     If 1, it is sensitive to rising edges of the input.
+
+                     This bit may only be changed while the `enable` bit is `0`.
+                  '''
+          },
+          { bits: "15:8",
+            name: "input_select",
+            resval: "0",
+            desc: '''Index of the input that this period counter should sample.
+                     The value must be smaller than the number of inputs minus one.
+
+                     This may only be changed while the `enable` bit is `0`.
+                  '''
+          },
+          { bits: "23:16",
```

Comment:
Can we switch `prescaler` and `input_select` ? That way there is no layout change if the number of GPIOs change.

Reply:
8 bits allows selecting up to 256 GPIOs; isn't that enough for the foreseeable future? I was thinking it would be more likely that the prescaler needs to become wider.

Reply:
oh makes sense. I stubbornly whought it's just log2(N=32) wide.